test/parallel/test-http-timeout-client-warning.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+const common = require('../common');
+const http = require('http');
+const assert = require('assert');
+
+// Checks that the setTimeout duration overflow warning is emitted
+// synchronously and therefore contains a meaningful stacktrace.
+
+process.on('warning', common.mustCall((warning) => {
+  assert(warning.stack.includes(__filename));
+}));
+
+const server = http.createServer((req, resp) => resp.end());
+server.listen(common.mustCall(() => {
+  http.request(`http://localhost:${server.address().port}`)
+    .setTimeout(2 ** 40)
+    .end(() => {
+      server.close();
```

Comment:
I don't know if this would be considered an issue with the test... there's a race condition between the server being closed and the request being processed. Most of these `http` tests would put the `server.close();` in an event listener for `end` rather than a callback. Not sure if this is expected behaviour or a bug though... I can see a case for both.

Reply:
See something like https://github.com/nodejs/node/blob/master/test/parallel/test-http-client-upload.js for an example.